waiter/src/waiter/core.clj

```diff
@@ -623,17 +627,18 @@
                                                  router-id async-request-store-atom make-http-request-fn instance-rpc-chan service-id metric-group
                                                  instance reason-map request-properties location response-headers)))
    :prepend-waiter-url (pc/fnk [[:settings port hostname]]
-                         (fn [endpoint-url]
-                           (if (str/blank? endpoint-url)
-                             endpoint-url
-                             (str "http://" hostname ":" port endpoint-url))))
+                         (let [hostname (if (sequential? hostname) (first hostname) hostname)]
```

Comment:
The sequence can be empty, I think we need to enforce in the `settings` schema that the list may not be empty.

Reply:
Done